types/bagpipes/index.d.ts

```diff
@@ -0,0 +1,178 @@
+// Type definitions for bagpipes 0.1
+// Project: https://github.com/apigee-127/bagpipes
+// Definitions by: Michael Mrowetz <https://github.com/micmro>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface FittingContext {
+    /** The input defined in the fitting definition
+     * (string, number, object, array)
+     */
+    input: any;
+    /** Output to be delivered to the next fitting or client */
+    output: any;
+    [prop: string]: any;
+}
+
+/**
+ * Fitting types has pre-defined fittings `system` and `user`
+ * but can be any any string for custom types like `swagger` or `node-machine`
+ */
+type FittingType = "system" | "user" | string;
+
+interface FittingDef {
+    /**
+     * If type is omitted (as it must be for in-line usage), Bagpipes will
+     * first check the user fittings then the system fittings for the name and
+     * use the first fitting found.
+     *
+     * Thus be aware that if you define a fitting with the same name as a
+     * system one, your fitting will override it.
+     */
+    type?: FittingType;
+    /** The name of the fitting of the type specified */
+    name?: string;
+    /** Static values passed to the fitting during construction */
+    config?: any;
+    /** Dynamic values passed to the fitting during execution */
+    input?: any;
+    /** The name of the context key to which the output value is assigned */
+    output?: any;
+    // allow other config settings
+    [prop: string]: any;
+}
+
+/** Middleware style `next` callback  */
+type Callback = (err: Error | null | undefined, res?: any) => void;
```

Comment:
I've inlined it but `res` is based on external inputs (from `input`), so it could be anything.